nonprimary_foreignkey/fields.py

```diff
@@ -0,0 +1,111 @@
+import django
+from django.core import checks
+from django.utils.functional import cached_property
+
+if django.VERSION < (1, 8):
+    from django.db.models.loading import get_model
+else:
+    from django.apps import apps
+    get_model = apps.get_model
+
+
+UNDEFINED = object()
+
+
+class NonPrimaryForeignKey(object):
+    """
+    Descriptor class for handling non-primary foreign keys.
+
+    The implementation is loosely based on ``GenericForeignKey`` from the
+    ``django.contrib.contenttypes`` module.
+    """
+
+    def __init__(self, to_model, from_field, to_field):
```

Comment:
Might be nice to allow `to_field=None` and default to the name of the `from_field`.